Networking/Networking/Remote/InAppPurchasesRemote.swift

```diff
@@ -90,6 +89,19 @@ public extension InAppPurchasesRemote {
         }
     }
 
+    /// Checks the WPCOM billing system for whether or not an In-app Purchase transaction has been handled
+    ///
+    ///- Returns:The siteID for which the transactionID belongs to
+    ///
+    @MainActor
```

Comment:
Just wondering, why do we need this to be run on the Main Actor?

Reply:
We do! Otherwise we'll crash when calling the remote inside `Transactions.Updates` if we happen to have anything in the queue waiting to be processed.

Reply:
Oh that's interesting, do you have more information about that crash or documentation from Apple where they talk about using the main thread for this logic?
Even with that, I would decouple the need for execution on the main thread from the function itself, moving it to the function call, something like

`Task { @MainActor in ...
`

That way we can call this function from a background thread if necessary. Even if this is not the case because we only call it there, it's a good practice to decouple and **only** use @MainActor if the function really needs to execute in the main thread, e.g. UI actions.

Reply:
> Oh that's interesting, do you have more information about that crash or documentation from Apple where they talk about using the main thread for this logic?

As far as I can see this bit is not on Apple's end but on ours when dealing with authenticated requests to wpcom:

The crash seems to happens because when we call the remote, there's a call to `WKWebView` that doesn't happen in the main thread, which causes the fatal:

When we create and enqueue the network request as `DotComRequest`, this is constructed by injecting the WPCOM token into an `AuthenticatedDotcomRequest`, which uses the WooCommerce iOS `UserAgent`, which calls a WKWebView instance. And this can only be used from the main thread:

```
-[WKWebView .cxx_construct] must be used from main thread only
```

> Even with that, I would decouple the need for execution on the main thread from the function itself, moving it to the function call

Good call, I moved this to a different method and moved the main thread logic to it. Here: https://github.com/woocommerce/woocommerce-ios/pull/10247/commits/ab3bf022ed87b5ae4a8f6c1cd1a84ccecdb83bda

I think there's room for improvement on the latest changes as well, since we can get the transaction from the result, but that would involve further refactoring when dealing with the different result cases, so I left it outside of this PR for the moment, will open a separate enhancement request for improving that.

Reply:
Tracked follow-up issue here: https://github.com/woocommerce/woocommerce-ios/issues/10294